apps/api/src/queries/digital-text/digital-text-queries.e2e.spec.ts

```diff
@@ -0,0 +1,295 @@
+                    );
+
+                    expect(res.status).toBe(HttpStatusCode.notFound);
+                });
+            });
+
+            describe(`fetch many`, () => {
+                it(`should only return published digital texts`, async () => {
+                    await app.get(ArangoEventRepository).appendEvents(eventHistoryForMany);
+
+                    const res = await request(app.getHttpServer()).get(indexEndpoint);
+
+                    expect(res.status).toBe(httpStatusCodes.ok);
+                });
+            });
+        });
+
+        describe(`when the user is authenticated`, () => {
+            beforeAll(async () => {
+                ({ app, testRepositoryProvider, databaseProvider } = await setUpIntegrationTest(
+                    {
+                        ARANGO_DB_NAME: testDatabaseName,
+                    },
+                    {
+                        testUserWithGroups: dummyUserWithGroups,
+                    }
+                ));
+            });
+
+            describe(`fetch single (by ID)`, () => {
+                describe(`when there are no existing digital texts`, () => {
+                    it(`should return not found`, async () => {
+                        const res = await request(app.getHttpServer()).get(
+                            buildDetailEndpoint(buildDummyUuid(456))
+                        );
+
+                        expect(res.status).toBe(httpStatusCodes.notFound);
+                    });
+                });
+
+                describe(`when there is a digital text with the given ID`, () => {
+                    describe(`when the digital text is published`, () => {
+                        it(`should return the corresponding result`, async () => {
+                            await app.get(ArangoEventRepository).appendEvents(validEventHistory);
+
+                            const res = await request(app.getHttpServer()).get(
+                                buildDetailEndpoint(digitalTextId)
+                            );
+
+                            const { body } = res;
+
+                            const result = body as DigitalTextViewModel;
+
+                            expect(result.id).toBe(digitalTextId);
+
+                            const expectedTitle = buildMultilingualTextWithSingleItem(
+                                creationEvent.payload.title,
+                                creationEvent.payload.languageCodeForTitle
+                            ).toDTO();
+
+                            expect(result.title).toEqual(expectedTitle);
+                        });
+                    });
+
+                    describe(`when the digital text is not published`, () => {
+                        describe(`when the user is authenticated`, () => {
+                            describe(`when the user is not part of the digital text's ACL`, () => {
+                                it(`should return not found`, async () => {
```

Comment:
Should we have a message that distinguishes that a resource exists but the user does not have the permission to see it?  or is that way too much additional code?

Reply:
No. We do not want to provide users information about which resources exist or do not exist unless they have permission. That's why we return not found. I suppose it's not real security since it's just obscurity, but I still prefer it this way.